contracts/asset-proxy/contracts/src/bridges/CurveBridge.sol

```diff
@@ -55,32 +63,32 @@ contract CurveBridge is
         returns (bytes4 success)
     {
         // Decode the bridge data to get the Curve metadata.
-        (address curveAddress, int128 fromCoinIdx, int128 toCoinIdx, int128 version) = abi.decode(bridgeData, (address, int128, int128, int128));
-        ICurve exchange = ICurve(curveAddress);
+        CurveBridgeData memory data = abi.decode(bridgeData, (CurveBridgeData));
 
-        address fromTokenAddress = exchange.underlying_coins(fromCoinIdx);
+        address fromTokenAddress = ICurve(data.curveAddress).underlying_coins(data.fromCoinIdx);
         require(toTokenAddress != fromTokenAddress, "CurveBridge/INVALID_PAIR");
+        uint256 fromTokenBalance = IERC20Token(fromTokenAddress).balanceOf(address(this));
```

Comment:
Any reason not to use `LibERC20Token` here?

Reply:
`LibERC20Token` is too permissive with `balanceOf/allowance` (won't propagate reverts). Those were created basically for DevUtils' sake.